portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -87,40 +91,44 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     // Convert List of comments to JSON.
     Gson gson = new Gson();
-    String commentsListJson = gson.toJson(commentsList);
     response.setContentType("application/json;");
-    response.getWriter().println(commentsListJson);
+    response.getWriter().println(gson.toJson(commentsList));
   }
 
   /**
-   * Processes HTTP POST requests for the /data servlet The requests are responded to by appending
-   * the 'comment-text' argument of the POST request to the Datastore database. The client is then
-   * redirected back to the com.html page.
+   * Processes HTTP POST requests for the /data servlet The requests are responded to by appending a
+   * comment to the datastore database. All necessary fields will be assembled here. If the user is
+   * not logged in, then the status field in the response json will be set to 'no-login'
    *
    * @param request Information about the POST Request
    * @param response Information about the servlet's response
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Constructs comment and gets comment sentiment
-    String commentText = getParameter(request, "comment-text", "");
-    float commentSentiment = getSentiment(commentText);
+    UserService userService = UserServiceFactory.getUserService();
 
-    Comment comment = new Comment(commentText, commentSentiment);
+    if (userService.isUserLoggedIn()) {
```

Comment:
nit: the other order would be slightly cleaner. e.g.

```
if (! userService.isUserLoggedIn()) {
 ...
 return;
}
// logged in code dedented
```

Reply:
Will change